server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -503,7 +505,23 @@ protected void doClose() {
         keepAliveReaper.cancel();
     }
 
+    // Wraps the listener to avoid sending StackTrace if error_trace is set to false
+    private <T> ActionListener<T> maybeWrapListenerForStackTrace(ActionListener<T> listener) {
+        String header = threadPool.getThreadContext().getHeaderOrDefault("error_trace", "true");
```

Comment:
Can you remind me the rationale behind using true as a default value here?

Reply:
We wanted the node behaviour to be consistent across version. And by default we return the stack trace.
If this is no longer a concern I can use the ERROR_CONTEXT_DEFAULT here too.

Reply:
Ok, so the idea is that nodes on older version don't have a way to request the stacktrace, hence if we remove it by default they are never going to get it even if error_trace is true. is that it? If so can you clarify that in a comment perhaps?

Reply:
Updated comment and gh description